sdk/provisioning/Azure.Provisioning/src/Primitives/Resource.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Azure.Core;
+using Azure.Provisioning.Expressions;
+
+namespace Azure.Provisioning.Primitives;
+
+/// <summary>
+/// Represents an Azure resource.
+/// </summary>
+/// <param name="resourceName">The friendly namee of the resource.</param>
+/// <param name="resourceType">The type of the resource.</param>
+/// <param name="resourceVersion">The version of the resource.</param>
+/// <param name="context">Optional ProvisioningContext.</param>
+public abstract class Resource(string resourceName, ResourceType resourceType, string? resourceVersion = default, ProvisioningContext? context = default)
+    : NamedProvisioningConstruct(resourceName, context)
+{
+    /// <summary>
+    /// Gets the type of the resource.
+    /// </summary>
+    public ResourceType ResourceType { get; private set; } = resourceType;
+
+    /// <summary>
+    /// Gets or sets the version of the resource.
+    /// </summary>
+    public string? ResourceVersion { get; set; } = resourceVersion;
+
+    /// <summary>
+    /// Gets whether this is referencing an existing resource or we're defining
+    /// a new resource.
+    /// </summary>
+    public bool IsExistingResource { get; protected set; } = false;
+
+    /// <summary>
+    /// Compose the resource into a provisioning plan that can be saved as
+    /// Bicep or deployed directly.
+    /// </summary>
+    /// <param name="context">Optional ProvisioningContext.</param>
+    /// <returns>
+    /// A provisioning plan that can be saved as Bicep or deployed directly.
+    /// </returns>
+    public virtual ProvisioningPlan Build(ProvisioningContext? context = default)
+    {
+        context ??= DefaultProvisioningContext;
+        return context.DefaultInfrastructure.Build(context);
+    }
```

Comment:
Calling `.Build()` on a `Resource` or `Infrastructure` to get a `ProvisioningPlan`